zhmccli/_cmd_user_pattern.py

```diff
@@ -0,0 +1,674 @@
+
+
+@user_pattern_group.command('show', options_metavar=COMMAND_OPTIONS_METAVAR)
+@click.argument('USER_PATTERN', type=str, metavar='USER_PATTERN')
+@click.pass_obj
+def user_pattern_show(cmd_ctx, user_pattern):
+    """
+    Show the details of an HMC user pattern.
+
+    The following properties are shown in addition to those returned by the HMC:
+
+    \b
+      - 'parent-name' - Name of the parent Console.
+      - 'specific-template-name' -
+        Name of the user template referenced by
+        'specific-template-uri'.
+      - 'template-name-override-ldap-server-definition-name' -
+        Name of the LDAP Server Definition referenced by
+        'template-name-override-ldap-server-definition-uri'.
+      - 'template-name-override-default-template-name' -
+        Name of the user template referenced by
+        'template-name-override-default-template-uri'.
+      - 'ldap-group-ldap-server-definition-name' -
+        Name of the LDAP Server Definition referenced by
+        'ldap-group-ldap-server-definition-uri'.
+      - 'ldap-group-default-template-name' -
+        Name of the user template referenced by
+        'ldap-group-default-template-uri'.
+      - 'domain-name-restrictions-ldap-server-definition-name' -
+        Name of the LDAP Server Definition referenced by
+        'domain-name-restrictions-ldap-server-definition-uri'.
+
+    The following old properties are shown that are set by the HMC to be
+    consistent with the preferred new properties:
+
+    \b
+      - 'ldap-server-definition-name' -
+        Name of the LDAP Server Definition referenced by
+        'ldap-server-definition-uri'.
+      - 'user-template-name' -
+        Name of the specific user template referenced by
+        'user-template-uri'.
+
+    In addition to the command-specific options shown in this help text, the
+    general options (see 'zhmc --help') can also be specified right after the
+    'zhmc' command name.
+    """
+    cmd_ctx.execute_cmd(lambda: cmd_user_pattern_show(cmd_ctx, user_pattern))
+
+
+@user_pattern_group.command('create', options_metavar=COMMAND_OPTIONS_METAVAR)
+@click.option('--name', type=str, required=True,
+              help='The name of the new user pattern.')
+@click.option('--description', type=str, required=False,
+              help='The description of the new user pattern.')
+@click.option('--pattern', type=str, required=True,
+              help='The pattern for matching userids. '
+              'The combination of "pattern" and "type" must be unique.')
+@click.option('--type', type=click.Choice(['glob-like', 'regular-expression']),
+              required=True,
+              help='The style in which the pattern is expressed.')
+@click.option('--retention-time', type=int, required=False, default=1,
+              help='The time in days that the pattern-based users created '
+              'based on this user pattern will be retained after being '
```

Comment:
So one can concat strings without an operator between? Interesting.

Reply:
Yes. It even has precedence over the dot operator, so you can do `"{}"\n"xx".format(value)`